packages/create-catalyst/src/utils/checkout-ref.ts

```diff
@@ -5,7 +5,7 @@ import { isExecException } from './is-exec-exception';
 export function checkoutRef(repoDir: string, ref: string): void {
   try {
     // Attempt to checkout the specified ref
-    execSync(`git checkout ${ref}`, {
+    execSync(`git checkout -- ${ref}`, {
```

Comment:
Chatted offline, the part after `--` is supposed to be a filename to checkout on a branch. e.g. `git checkout <some-branch> -- <some-file>`